### PR TITLE
Add ServiceByHcPartyHealthElementIdsFilter to filter services by health elements

### PR DESCRIPTION
**Pull request by lisatr2812**

### CODE REVIEW

File: src/main/kotlin/org/taktik/icure/asyncdao/impl/ContactDAOImpl.kt [hunk 1]
```diff
@@ -337,15 +337,17 @@ class ContactDAOImpl(couchDbProperties: CouchDbProperties,
     }
 
 
-    @View(name = "service_id_by_health_element", map = "classpath:js/contact/Service_id_by_health_element_id.js")
-    override fun listServiceIdsForHealthElementId(healthElementId: String) = flow {
+    @View(name = "service_id_by_health_elements", map = "classpath:js/contact/Service_id_by_hcparty_helement_ids.js")
```
Comment (LotuxPunk) [inline src/main/kotlin/org/taktik/icure/asyncdao/impl/ContactDAOImpl.kt:340]:
```suggestion
    @View(name = "service_ids_by_health_elements", map = "classpath:js/contact/Service_ids_by_hcparty_helement_ids.js")
```

File: src/main/kotlin/org/taktik/icure/asynclogic/ContactLogic.kt [hunk 2]
```diff
@@ -60,7 +60,8 @@ interface ContactLogic : EntityPersister<Contact, String> {
     fun listContactsByHcPartyServiceId(hcPartyId: String, formId: String): Flow<Contact>
     fun listContactsByExternalId(externalId: String): Flow<Contact>
     fun listServicesByAssociationId(associationId: String): Flow<Service>
-    fun listServicesForHealthElementId(healthElementId: String): Flow<Service>
+    fun listServicesForHealthElementIds(hcPartyId: String, healthElementIds: List<String>): Flow<Service>
```
Comment (LotuxPunk) [inline src/main/kotlin/org/taktik/icure/asynclogic/ContactLogic.kt:63]:
```suggestion
    fun listServicesByHealthElementIds(hcPartyId: String, healthElementIds: List<String>): Flow<Service>
```
To keep consistency with the DAO

File: src/main/kotlin/org/taktik/icure/asynclogic/ContactLogic.kt [hunk 3]
```diff
@@ -60,7 +60,8 @@ interface ContactLogic : EntityPersister<Contact, String> {
...
+    fun listServiceIdsForHealthElementIds(hcPartyId: String, healthElementIds: List<String>): Flow<String>
```
Comment (LotuxPunk) [inline src/main/kotlin/org/taktik/icure/asynclogic/ContactLogic.kt:64]:
```suggestion
    fun listServiceIdsByHealthElementIds(hcPartyId: String, healthElementIds: List<String>): Flow<String>
```
To keep consistency with the DAO

File: src/main/kotlin/org/taktik/icure/services/external/rest/v2/dto/filter/service/ServiceByHcPartyHealthElementIdsFilter.kt [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import org.taktik.icure.handlers.JsonPolymorphismRoot
+import org.taktik.icure.services.external.rest.v2.dto.filter.AbstractFilterDto
+
+@JsonPolymorphismRoot(AbstractFilterDto::class)
+@JsonDeserialize(using = JsonDeserializer.None::class)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@KotlinBuilder
+data class ServiceByHcPartyHealthElementIdsFilter(
+        override val desc: String?,
```
Comment (LotuxPunk) [inline src/main/kotlin/org/taktik/icure/services/external/rest/v2/dto/filter/service/ServiceByHcPartyHealthElementIdsFilter.kt:19]:
```suggestion
        override val desc: String? = null,
```

**Review by LotuxPunk — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/taktik/icure/asyncdao/impl/ContactDAOImpl.kt (hunk 1 above)
Comment (lisatr2812) [reply src/main/kotlin/org/taktik/icure/asyncdao/impl/ContactDAOImpl.kt:340]: The other views are named using single form and not plural as well : I will follow the same pattern

**Pull request merged**